src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+  createConcurrentSafePromise,
+  addQueryID,
+  debounce,
+  debounceAsync,
+  addAbsolutePosition,
+  noop,
+  escapeHits,
+} from '../../lib/utils';
+import { Connector, Hits, Hit, FindAnswersResponse } from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'answers',
+  connector: true,
+});
+
+export type AnswersRendererOptions = {
+  /**
+   * The matched hits from Algolia API.
+   */
+  hits: Hits;
+
+  /**
+   * whether it's still loading the results from the /answers API
+   */
+  isLoading: boolean;
+};
+
+export type AnswersConnectorParams = {
+  /**
+   * Attributes to use for predictions.
+   * If empty, we use all `searchableAttributes` to find answers.
+   * All your `attributesForPrediction` must be part of your `searchableAttributes`.
+   */
+  attributesForPrediction?: string[];
+
+  /**
+   * The languages in the query. Currently only supports `en`.
+   */
+  queryLanguages: Array<'en'>;
+
+  /**
+   * Maximum number of answers to retrieve from the Answers Engine.
+   * Cannot be greater than 1000.
+   */
+  nbHits?: number;
+
+  /**
+   * Debounce time in milliseconds to debounce render
+   * default: 100
+   */
+  renderDebounceTime?: number;
+
+  /**
+   * Debounce time in milliseconds to debounce search
+   * default: 100
+   */
+  searchDebounceTime?: number;
+
+  /**
+   * Whether to escape HTML tags from hits string values.
+   *
+   * @default true
+   */
+  escapeHTML?: boolean;
+};
+
+export type AnswersConnector = Connector<
+  AnswersRendererOptions,
+  AnswersConnectorParams
+>;
+
+const connectAnswers: AnswersConnector = function connectAnswers(
+  renderFn,
+  unmountFn = noop
+) {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const {
+      queryLanguages,
+      attributesForPrediction,
+      nbHits = 1,
+      renderDebounceTime = 100,
+      searchDebounceTime = 100,
+      escapeHTML = true,
+    } = widgetParams || ({} as typeof widgetParams);
+
+    if (!queryLanguages || queryLanguages.length === 0) {
+      throw new Error(
+        withUsage('The `queryLanguages` expects an array of strings.')
+      );
+    }
+
+    const runConcurrentSafePromise = createConcurrentSafePromise<
+      FindAnswersResponse<Hit>
+    >();
+
+    let lastResult: Partial<FindAnswersResponse<Hit>>;
+    let isLoading = false;
+    const debouncedRender = debounce(renderFn, renderDebounceTime);
+    let debouncedRefine;
+
+    return {
+      $$type: 'ais.answers',
+
+      init(initOptions) {
+        const { state, instantSearchInstance } = initOptions;
+        const answersIndex = instantSearchInstance.client!.initIndex!(
+          state.index
+        );
```

Comment:
this can be done in widget creation, before init

Reply:
@Haroenv do you mean like in line 105? We need an access to `instantSearchInstance` and I don't see any way before init.